src/useThreadIndent.tsx

```diff
@@ -7,28 +7,25 @@ const INDENT_SIZE = ICON_SIZE / 2 - VERTICAL_LINE_SIZE
 
 export type CssUnit = 'px' | 'rem' | 'em' | 'vw' | 'vh' | 'vmin' | 'vmax'
 
-interface IndentSizes {
-  iconSize: number
-  margin: number
+type CssUnitValue = `${number}${CssUnit}`
+export interface IndentSizes {
+  iconSize: CssUnitValue
+  margin: CssUnitValue
 }
 
-function useThreadIndent(): { [P in keyof IndentSizes]: `${number}${CssUnit}` } {
+function useThreadIndent(): IndentSizes {
   const { isMobile } = useScreenSize()
-  const [{ iconSize, margin }, setIndent] = useState<IndentSizes>({
-    iconSize: isMobile ? ICON_SIZE / 2 : ICON_SIZE,
-    margin: isMobile ? (INDENT_SIZE / 2) : INDENT_SIZE,
-  })
-
-  useEffect(() => {
-    setIndent({
-      iconSize: isMobile ? ICON_SIZE / 2 : ICON_SIZE,
-      margin: isMobile ? (INDENT_SIZE / 2) : INDENT_SIZE,
-    })
-  }, [isMobile])
+  const [{ iconSize, margin }, setIndent] = useState<IndentSizes>(getIndentSizes(isMobile))
 
+  useEffect(() => setIndent(getIndentSizes(isMobile)), [isMobile])
```

Comment:
When calculating derived value based on another value, do not use `useEffect` — use `useMemo` instead.